### ISSUE TITLE
[Issue] `regctl artifact list` doesn't list all the artifacts

### ISSUE THREAD

**Issue by Vishal-Chdhry** [regclient/regclient#417]
### Current Behavior
`regctl` doesnt list all the referrers
```bash
regctl artifact list jimnotarytest.azurecr.io/jim/net-monitor:v1                                                             

Subject:   jimnotarytest.azurecr.io/jim/net-monitor:v1

Referrers:
```

### Expected Behavior
It should list all artifacts like here
```bash
oras discover -o tree jimnotarytest.azurecr.io/jim/net-monitor:v1            
jimnotarytest.azurecr.io/jim/net-monitor:v1
├── vuln-scan
│   └── sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8
│       └── application/vnd.cncf.notary.signature
│           └── sha256:22aed3423fda62d88a09c47cbf1e4ca0441376fbc0270cafb193d4b9146eedc4
└── application/vnd.cncf.notary.signature
    └── sha256:e9277c9696d262e699583b7f4304eb9a3e7899a0d6c6b3c6b88327926347aaec
```

### Steps To Reproduce

Run this command `regctl artifact list jimnotarytest.azurecr.io/jim/net-monitor:v1`

### Version

```text
regctl version                                                                                              
VCSRef:     f43ada160d953317c51e7d9170afa2a316f4b9be
VCSCommit:  f43ada160d953317c51e7d9170afa2a316f4b9be
VCSState:   clean
VCSDate:    2023-02-26T02:43:41Z
Platform:   darwin/arm64
GoVer:      go1.20.1
GoCompiler: gc
(paste output of `regctl version` or similar for binaries, or `docker image inspect regclient/...` on the image and paste the labels)
```

### Environment

- Running as binary or container:
- Host platform: Mac
- Registry description: ACR

**Comment by sudo-bmitch** [comment]
`regctl artifact list` quickly shows artifacts for a single manifest. `regctl artifact tree` recursively lists artifacts, including the child manifests, and referrers to those artifacts. Give the latter a try and let me know if that resolves the issue.

**Comment by Vishal-Chdhry** [comment]
@sudo-bmitch This is what `regclt artifact tree` returns
```
regctl artifact tree jimnotarytest.azurecr.io/jim/net-monitor:v1                                                                     
Ref: jimnotarytest.azurecr.io/jim/net-monitor:v1
Digest: sha256:ba7000206594c2d72c3ab550453004c0dc50961157e5ebd2fb8ea1890099d02d
```
It doesn't list everything. My issue is that I want to use refererrList from regclient, but it has the same issue as well https://github.com/regclient/regclient/blob/7b09bbe2da8bfd92cd3df3e726390396faa2a420/referrer.go#L12

**Comment by sudo-bmitch** [comment]
Got it. I see at least one issue with the manifest get on this. They are using `application/vnd.oci.artifact.manifest.v1+json` which is no longer part of the OCI spec, and I need to add some headers since there's still support for reading this data in regclient. I'll dig through the ORAS and regclient code to see what's happening with the referrers API on ACR.

**Comment by Vishal-Chdhry** [comment]
Thank you so much @sudo-bmitch!

Also there is one more thing, I get this error when I fetch the manifest of the `vuln-scan` artifact as follows
```bash
regctl manifest get --format body jimnotarytest.azurecr.io/jim/net-monitor@sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8 | jq .
failed to get manifest jimnotarytest.azurecr.io/jim/net-monitor@sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8: request failed: not found [http 404]: {"errors":[{"code":"MANIFEST_UNKNOWN","message":"OCI artifact found, but accept header does not support OCI artifacts"}]}
```

Here is the referrers list
```bash
oras discover -o tree jimnotarytest.azurecr.io/jim/net-monitor:v1 
jimnotarytest.azurecr.io/jim/net-monitor:v1
├── vuln-scan
│   └── sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8
│       └── application/vnd.cncf.notary.signature
│           └── sha256:22aed3423fda62d88a09c47cbf1e4ca0441376fbc0270cafb193d4b9146eedc4
└── application/vnd.cncf.notary.signature
    └── sha256:e9277c9696d262e699583b7f4304eb9a3e7899a0d6c6b3c6b88327926347aaec
```

Here is the manifest
```bash
oras manifest fetch jimnotarytest.azurecr.io/jim/net-monitor@sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8 | jq .
{
  "mediaType": "application/vnd.oci.artifact.manifest.v1+json",
  "artifactType": "vuln-scan",
  "blobs": [
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "digest": "sha256:bf9500c04c507cc5074cd4b174aa0a1d78009b3d71c6d487257017da1bd01700",
      "size": 178782,
      "annotations": {
        "org.opencontainers.image.title": "scan.json"
      }
    }
  ],
  "subject": {
    "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
    "digest": "sha256:ba7000206594c2d72c3ab550453004c0dc50961157e5ebd2fb8ea1890099d02d",
    "size": 528
  },
  "annotations": {
    "org.opencontainers.artifact.created": "2023-04-19T15:46:25Z"
  }
}
```

I might be happening because the `Accept` header doesnt have `application/vnd.oci.artifact.manifest.v1+json` media type in it. (gcr crane has a similar issue as well and that gives the same error as well)

**Comment by sudo-bmitch** [comment]
Thanks for the report @Vishal-Chdhry. With the fix in #418, I'm now able to see the following:

```
$ regctl artifact list jimnotarytest.azurecr.io/jim/net-monitor:v1
Subject:                                      jimnotarytest.azurecr.io/jim/net-monitor:v1

Referrers:

  Name:                                       jimnotarytest.azurecr.io/jim/net-monitor@sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8
  Digest:                                     sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8
  MediaType:                                  application/vnd.oci.artifact.manifest.v1+json
  ArtifactType:                               vuln-scan
  Size:                                       561B
  Annotations:
    org.opencontainers.artifact.created:      2023-04-19T15:46:25Z

  Name:                                       jimnotarytest.azurecr.io/jim/net-monitor@sha256:e9277c9696d262e699583b7f4304eb9a3e7899a0d6c6b3c6b88327926347aaec
  Digest:                                     sha256:e9277c9696d262e699583b7f4304eb9a3e7899a0d6c6b3c6b88327926347aaec
  MediaType:                                  application/vnd.oci.image.manifest.v1+json
  ArtifactType:                               application/vnd.cncf.notary.signature
  Annotations:
    io.cncf.notary.x509chain.thumbprint#S256: ["855d6bb94c8292aeaf0cee2e1db844445f0145a105bcdf6c9cfb9699f70df7fc"]
    org.opencontainers.image.created:         2023-04-19T08:20:10Z

$ regctl artifact tree jimnotarytest.azurecr.io/jim/net-monitor:v1
Ref: jimnotarytest.azurecr.io/jim/net-monitor:v1
Digest: sha256:ba7000206594c2d72c3ab550453004c0dc50961157e5ebd2fb8ea1890099d02d
Referrers:
  - sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8: vuln-scan
    Referrers:
      - sha256:22aed3423fda62d88a09c47cbf1e4ca0441376fbc0270cafb193d4b9146eedc4: application/vnd.cncf.notary.signature
  - sha256:e9277c9696d262e699583b7f4304eb9a3e7899a0d6c6b3c6b88327926347aaec: application/vnd.cncf.notary.signature

$ regctl manifest get  jimnotarytest.azurecr.io/jim/net-monitor@sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8
Name:                                  jimnotarytest.azurecr.io/jim/net-monitor@sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8
MediaType:                             application/vnd.oci.artifact.manifest.v1+json
ArtifactType:                          vuln-scan
Digest:                                sha256:10c0b24faa551466b708d1677694eb65bbe4679ba10c3a5290ecec2e4f0af6c8
Annotations:
  org.opencontainers.artifact.created: 2023-04-19T15:46:25Z
Total Size:                            178.8kB

Blobs:

  Digest:                              sha256:bf9500c04c507cc5074cd4b174aa0a1d78009b3d71c6d487257017da1bd01700
  MediaType:                           application/vnd.oci.image.layer.v1.tar
  Size:                                178.8kB
  Annotations:
    org.opencontainers.image.title:    scan.json

Subject:
  Digest:                              sha256:ba7000206594c2d72c3ab550453004c0dc50961157e5ebd2fb8ea1890099d02d
  MediaType:                           application/vnd.docker.distribution.manifest.v2+json
```

If you experience any further issues with this, don't hesitate to let me know.
